server/routes/serviceProviderReferrals/editSessionPresenter.ts

```diff
@@ -45,5 +45,39 @@ export default class EditSessionPresenter {
       'duration',
       this.validationError
     ),
+    meetingMethod: this.chosenAppointmentDeliveryType(this.appointment.appointmentDeliveryType),
+    addressLine1: this.getAddressFromIndex(0, this.appointment.appointmentDeliveryAddress),
+    addressLine2: this.getAddressFromIndex(1, this.appointment.appointmentDeliveryAddress),
+    addressTownOrCity: this.getAddressFromIndex(2, this.appointment.appointmentDeliveryAddress),
+    addressCounty: this.getAddressFromIndex(3, this.appointment.appointmentDeliveryAddress),
+    addressPostCode: this.getAddressFromIndex(4, this.appointment.appointmentDeliveryAddress),
+  }
+
+  private getAddressFromIndex(index: number, address: string[] | null): string | null {
```

Comment:
Would array destructuring work here instead of this function, e.g. `const [addressLine1, addressLine2, addressTownOrCity, addressCounty, addressPostcode] = this.appintment.appointmentDeliveryAddress`? Might make things a little neater, as it's not that obvious from the indexes what each value represents.

You could then possibly return these values from a function and use them above e.g.

```
private getAddress(address: string[] | null): Address {
  const [addressLine1, addressLine2, addressTownOrCity, addressCounty, addressPostcode] = this.appointment.appointmentDeliveryAddress`

  return { addressLine1, addressLine2, addressTownOrCity, addressCounty, addressPostcode }
}
```

Reply:
Sorry this was a draft PR. It has been corrected in the latest commits.